src/app/src/main/kotlin/org/vaccineimpact/api/app/ChunkedFileCache.kt

```diff
@@ -29,7 +30,14 @@ class ChunkedFileCache(private val flushInterval: Long = TimeUnit.HOURS.toMillis
     override fun put(item: ChunkedFile)
     {
         recycle()
-        memoryCache[item.uniqueIdentifier] = MutablePair(item, System.currentTimeMillis())
+        if (memoryCache[item.uniqueIdentifier] == null)
```

Comment:
This null check was carried out [here](https://github.com/vimc/montagu-api/pull/418/files#diff-7ad6d3cde953d76738a50c8a0f770406L199) but it could have changed between then and now. Effectively this just make the race condition tighter and manual testing indicates that it does work so is a good intermediate solution. A really robust solution will be to make an initial endpoint call that just initialises the cache before any file chunks are posted, which will require some front-end work as well.

Reply:
What about making that section in BurdenEstimateUploadController.kt from lines 206-219 a critical section? (So only one thread at a time can ask if cachedMetaData == null - and the second asker is blocked until the first one finishes that section?

Reply:
Eg (slightly coarsely), BurdenEstimateUploadController.kt :180
private fun getFileMetadata(): ChunkedFile
->
private synhronized getFileMetadata(): ChunkedFile

Reply:
Yes I think that would work